src/microbe_stage/Spawners.cs

```diff
@@ -84,6 +84,90 @@ public static void SpawnCellBurstEffect(IWorldSimulation worldSimulation, Vector
     /// <summary>
     ///   Spawns an agent projectile
     /// </summary>
+    public static EntityRecord SpawnIronProjectile(IWorldSimulation worldSimulation,
+        float amount, float lifetime, Vector3 location, Vector3 direction, float scale, Entity emitter)
+    {
+        var recorder = SpawnIronProjectileWithoutFinalizing(worldSimulation,
+            amount, lifetime, location, direction, scale, emitter, out var entity);
+
+        FinalizeEntitySpawn(recorder, worldSimulation);
+
+        return entity;
+    }
+
+    public static EntityCommandRecorder SpawnIronProjectileWithoutFinalizing(IWorldSimulation worldSimulation,
+        float amount, float lifetime, Vector3 location, Vector3 direction, float scale,
+        Entity emitter, out EntityRecord entity)
+    {
+        var recorder = worldSimulation.StartRecordingEntityCommands();
+
+        entity = SpawnIronProjectileWithoutFinalizing(worldSimulation, recorder, amount, lifetime,
+            location, direction, scale, emitter);
+
+        return recorder;
+    }
+
+    public static EntityRecord SpawnIronProjectileWithoutFinalizing(IWorldSimulation worldSimulation,
+        EntityCommandRecorder commandRecorder, float amount, float lifetime,
+        Vector3 location, Vector3 direction, float scale, Entity emitter)
+    {
+        var normalizedDirection = direction.Normalized();
```

Comment:
This seems to have a lot in common with oxytoxy emission. So again here I'd suggest making a common helper method that sets up everyhing except the SiderophoreProjectile versus oxytoxy properties, that'll reduce code duplication and any projectile fixes are automatically picked up by both places.

Reply:
If I want to make SpawnAgentProjectile with siderophore flag, what do I do with AgentProperties?

Reply:
Make a helper function that siderohone and agent projectiles calls, and leave out the AgentProperties etc. components that aren't used by both. That way the common part is neatly contained in a single method and just the differences are then in the separate siderohpone and agent spawn methods.

> So again here I'd suggest making a common helper method that sets up everyhing except the SiderophoreProjectile versus oxytoxy properties, that'll reduce code duplication and any projectile fixes are automatically picked up by both places.